src/components/infinite-data-table.jsx

```diff
@@ -0,0 +1,427 @@
+  const {
+    showRowNumbers, selectable, showHeader, data: inData, columns: inColumns,
+  } = props;
+  const data = showHeader ? [{}, ...inData] : [...inData];
+  let columns = [...inColumns];
+  if (showRowNumbers) {
+    columns = [NUMBER_COLUMN, ...columns];
+  }
+  if (selectable) {
+    columns = [SELECT_COLUMN, ...columns];
+  }
+  return <InfiniteDataTableCore {...props} data={data} columns={columns} />;
+};
+
+export class InfiniteDataTableCore extends Component {
+  cache = new CellMeasurerCache({
+    fixedWidth: true,
+    DEFAULT_HEIGHT,
+    MIN_HEIGHT,
+  });
+
+  constructor(props) {
+    super(props);
+    this.getColumnWidth = this.getColumnWidth.bind(this);
+    this.cellRenderer = this.cellRenderer.bind(this);
+    this.isRowLoaded = this.isRowLoaded.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { data, columns } = this.props;
+    const { data: prevData } = prevProps;
+    if (!serializableDeepAreEqual(data, prevData)) {
+      for (let rowIndex = prevData.length; rowIndex <= data.length + 1; rowIndex += 1) {
+        for (let columnIndex = 0; columnIndex < columns.length; columnIndex += 1) {
+          this.cache.clear(rowIndex, columnIndex);
+        }
+      }
+    }
+  }
+
+  getHeader() {
+    const { columns } = this.props;
+    return columns.map(column => ({ label: column.label || '', render: x => x }));
+  }
+
+  getColumnWidth({ index }) {
+    const { columns } = this.props;
+    const column = columns[index];
+    return column && column.width ? column.width : DEFAULT_WIDTH;
+  }
+
+  static getSortableHeaderCell({ column, onHeaderClick, style }) {
+    let className = 'table-head ';
+    if (column.sorted) {
+      className += column.sorted === 'ascend' ? 'table-header-ascend' : 'table-header-descend';
+    } else {
+      className += 'table-header-sortable';
+    }
+    return (
+      <button
+        style={style}
+        type="button"
+        key={column.name}
+        className={className}
+        onClick={() => onHeaderClick(column.name)}
+      >
+        {column.label || ''}
+      </button>
+    );
+  }
+
+  static getHeaderCell({ column, onHeaderClick, style }) {
+    if (column.sortable) {
+      return InfiniteDataTableCore.getSortableHeaderCell({ column, onHeaderClick, style });
+    }
+    return (
+      <div className="table-head table-header" style={style}>
+        {column.label || ''}
+      </div>
+    );
+  }
+
+  static getLoadingCell({ className, style }) {
+    return (
+      <div className={className} style={style}>
+        Loading...
+      </div>
+    );
+  }
+
+  static getSelectCell({
+    id, onSelect, style, className, selectedRows,
+  }) {
+    const checked = id in selectedRows;
+    return (
+      <div className={`${className} ${checked && 'table-row-selected'}`} style={style}>
+        <input type="checkbox" onChange={() => onSelect(id)} checked={id in selectedRows} />
+      </div>
+    );
+  }
+
+  static getNumberCell({
```

Comment:
Would a name like `getRowCountCell` be more explicit? I initially thought this was a function to render cells which had numbers in them.

Reply:
Done.